Framework/Algorithms/src/MergeRuns/SampleLogsBehaviour.cpp

```diff
@@ -0,0 +1,463 @@
+  StringTokenizer tokenizerTolerances(paramsTolerances, ",",
+                                      StringTokenizer::TOK_TRIM |
+                                          StringTokenizer::TOK_IGNORE_EMPTY);
+
+  auto tolerancesStringVector = tokenizerTolerances.asVector();
+
+  std::vector<double> tolerancesVector = createTolerancesVector(
+      tokenizer.asVector().size(), tolerancesStringVector, paramsTolerances);
+
+  StringTokenizer::Iterator i = tokenizer.begin();
+  std::vector<double>::iterator j = tolerancesVector.begin();
+
+  for (; i != tokenizer.end() && j != tolerancesVector.end(); ++i, ++j) {
+    auto item = *i;
+    auto tolerance = *j;
+
+    if (skipIfInPrimaryMap && (m_logMap.count(item) != 0)) {
+      continue;
+    }
+
+    if (map.count(item) != 0) {
+      throw std::invalid_argument(
+          "Error when making list of merge items, sample log \"" + item +
+          "\" defined more than once!");
+    }
+
+    std::shared_ptr<Property> prop;
+    try {
+      prop = std::shared_ptr<Property>(ws->getLog(item)->clone());
+    } catch (std::invalid_argument &) {
+      m_logger.warning() << "Could not merge sample log \"" << item
+                         << "\", does not exist in workspace!" << std::endl;
+      continue;
+    }
+
+    bool isNumeric;
+    double value = 0.0;
+    try {
+      value = ws->getLogAsSingleValue(item);
+      isNumeric = true;
+    } catch (std::invalid_argument &) {
+      isNumeric = false;
+      if (mergeType == time_series) {
+        m_logger.error() << item << " could not be converted to a numeric type"
+                         << std::endl;
+        continue;
+      }
+    }
+
+    if (mergeType == time_series) {
+      try {
+        // See if property exists already - merging an output of MergeRuns
+        prop = std::shared_ptr<Property>(
+            ws->getLog(item + TIME_SERIES_SUFFIX)->clone());
+      } catch (std::invalid_argument &) {
+        // Property does not already exist
+        std::unique_ptr<Kernel::TimeSeriesProperty<double>> timeSeriesProp(
+            new TimeSeriesProperty<double>(item + TIME_SERIES_SUFFIX));
+        std::string startTime = ws->mutableRun().startTime().toISO8601String();
+        timeSeriesProp->addValue(startTime, value);
+        ws->mutableRun().addLogData(
+            std::unique_ptr<Kernel::Property>(std::move(timeSeriesProp)));
+        prop = std::shared_ptr<Property>(
+            ws->getLog(item + TIME_SERIES_SUFFIX)->clone());
+      }
+    } else if (mergeType == list) {
+      try {
+        // See if property exists already - merging an output of MergeRuns
+        prop =
+            std::shared_ptr<Property>(ws->getLog(item + LIST_SUFFIX)->clone());
+      } catch (std::invalid_argument &) {
+        ws->mutableRun().addProperty(item + LIST_SUFFIX, prop->value());
+        prop =
+            std::shared_ptr<Property>(ws->getLog(item + LIST_SUFFIX)->clone());
+      }
+    }
```

Comment:
No, warn and fail do not require adding a new sample log, I have added a comment to this effect.